sympy/polys/series/tring.py

```diff
@@ -0,0 +1,89 @@
+from __future__ import annotations
+
+from sympy.external.gmpy import GROUND_TYPES, MPQ, MPZ
+
+from sympy.polys.domains.domain import Domain, Er, Ef
+from sympy.polys.series.base import PowerSeriesRingProto, PowerSeriesRingFieldProto
+from sympy.polys.series.ringpython import (
+    USeries,
+    PythonPowerSeriesRingZZ,
+    PythonPowerSeriesRingQQ,
+)
+
+from typing import Generic, TYPE_CHECKING, cast, Union
+
+
+if GROUND_TYPES == "flint":
+    from sympy.polys.series.ringflint import (
+        ZZSeries,
+        QQSeries,
+        FlintPowerSeriesRingZZ,
+        FlintPowerSeriesRingQQ,
+    )
+
+    flint = True
+else:
+    if TYPE_CHECKING:
+        from sympy.polys.series.ringflint import (
+            ZZSeries,
+            QQSeries,
+            FlintPowerSeriesRingZZ,
+            FlintPowerSeriesRingQQ,
+        )
+    else:
+        ZZSeries = QQSeries = None
+        FlintPowerSeriesRingZZ = FlintPowerSeriesRingQQ = None
+    flint = False
+
+
+class TSeriesElement(Generic[Er]):
+    """Dummy type for lower power series elements, used only for static type checking."""
+
+    # Dummy method to ensure that TSeriesElement is invariant in Er.
+    def __invalid__(self, other: Er) -> Er:
+        raise NotImplementedError
+
+    flint = True
+
+
+# Common type for the all lower power series elements.
+TElement = Union[USeries, ZZSeries, QQSeries]
```

Comment:
I don't think that `TElement` should be used anywhere. Everywhere else should use `TSeriesElement[Er]` (or `TSeriesElement[Ef]`).

The purpose of the `_power_series_ring` function below is to make this seem like a concrete type everywhere else so that the actual underlying types like `ZZSeries` are obscured behind the opaque `TSeriesElement` type.